navbar.ts

```diff
@@ -0,0 +1,433 @@
+        if (!(attrName === "menu") && !(attrName == "back"))
+          this.noParentContianer.push(anchor);
+      } else {
+        if (!this.parentContainer[attributes.pnt]) {
+          this.parentContainer[attributes.pnt] = [];
+        }
+        this.parentContainer[attributes.pnt].push(anchor);
+      }
+      i++;
+    }
+    
+    const toggle = this.querySelector(this.togglesel);
+    if (toggle) {
+      const originalMenu = toggle as HTMLAnchorElement;
+      this.menuElement = originalMenu.cloneNode(true) as HTMLAnchorElement;
+      originalMenu.style.display = "none";
+      document.body.appendChild(this.menuElement);
+    }
+
+    this.menuElement.style.display = "block";
+    this.menuElement.style.width = "100%";
+    this.menuElement.style.background = "blue";
+
+    let temps = document.getElementsByTagName("template");
+
+    Array.from(temps).forEach((template) => {
+      if (template.getAttribute("data-type") === "back") {
+        const newContent = template.content.cloneNode(true) as DocumentFragment;
+        this.backElement = document.createElement("a");
+        this.backElement.appendChild(newContent.cloneNode(true));
+      } else if (template.getAttribute("data-type") === "close") {
+        const newContent = template.content.cloneNode(true) as DocumentFragment;
+        this.closeElement = document.createElement("a");
+        this.closeElement.appendChild(newContent.cloneNode(true));
+      }
+    });
+
+    this.backElement.id = "back";
+    this.closeElement.id = "close";
+    this.parentContainer["menu"] = this.noParentContianer;
+    if (this.cutoffwidth === 0) {
+      this.cutoffwidth = this.getNavbarLength();
+    }
+
+    window.addEventListener("resize", this.updateNavbar.bind(this));
+    this.updateNavbar();
+
+    this.menuElement.onclick = this.handleMenu.bind(this);
+    this.closeElement.onclick = this.handleClose.bind(this);
+
+    this.menu = document.createElement("div");
+    if (this.mode == "overlay") {
+      this.menu.appendChild(this.overlayNode());
+    } else if (this.mode == "saccordion") {
+      this.menu.appendChild(this.accordionCreator());
+    } else {
+      this.menu.appendChild(this.accordionCreator("menu", false));
+    }
+    this.menu.style.display = "none";
+    document.body.appendChild(this.menu);
+
+    for (const key in this.overlayContainer) {
+      const anchor = this.getAnchorElement(key);
+      if (anchor) {
+        anchor.onclick = () => {
+          if (this.currentPath.length > 0) {
+            const previousAnchor =
+              this.currentPath[this.currentPath.length - 1];
+
+            const attributes = this.parseAttributes(
+              previousAnchor.getAttribute("attr") || ""
+            );
+            this.overlayContainer[attributes.nm][0].style.display = "none";
+          }
+          this.overlayContainer[key][0].style.display = "block";
+          this.currentPath.push(anchor);
+        };
+      }
+    }

```

Comment:
We are looping again here which can avoided if this operation is done in the `overlayCreator()` where you actually add the anchor name to the `this.overlayContainer`. I think the usage of `this.overlayContainer` can be entirely removed.